modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/side_panel_content/index.js

```diff
@@ -18,9 +18,15 @@ import {
 	OPEN_MODAL_FROM_IFRAME,
 } from 'frontend-taglib-clay/data_set_display/utils/eventsDefinitions';
 
+function handleOpenModalFromSidePanel(payload) {
+	window.top.Liferay.fire(OPEN_MODAL_FROM_IFRAME, payload);
+}
+
 function initializeSidePanelListeners() {
-	Liferay.on(OPEN_MODAL, (payload) => {
-		window.top.Liferay.fire(OPEN_MODAL_FROM_IFRAME, payload);
+	Liferay.on(OPEN_MODAL, handleOpenModalFromSidePanel);
+
+	Liferay.on('beforeNavigate', () => {
```

Comment:
Two problems here:
1. `beforeNavigate` is not good event for detaching listeners, because SPA can be turned off.
2. You are still left with `beforeNavigate`, which may cause bugs or memory leak.

Our typical pattern would be something like this:
```js
function handleDestroyPortlet() {
	Liferay.detach(OPEN_MODAL, handleOpenModalFromSidePanel)

	Liferay.detach('destroyPortlet', handleDestroyPortlet);
}

Liferay.on('destroyPortlet', handleDestroyPortlet);
```

Reply:
This seems to be wrapped in a `frontend:component` call, in which case, lifecycle is already handled and this would only need a `dispose` method for it to be cleaned up whenever necessary.

This uncovers a different problem, though. Using `frontend:component` will do a `new foo()` so it's better suited for Class-type components, so ideally this would be refactored a little bit so that it better matches a class-component approach if it is to be invoked through `frontend:component`.

Reply:
That code runs in side panel iframes and should be executed only in case Senna is active so it won't be a problem keeping in that way. Anyway, if the common pattern is using `destroyPortlet` (and it makes sense) I'll change it :)

Reply:
`beforeNavigate` implied the fact that sennaJs was active (being part of its lifecycle events). The only `fire` I've found is infact [here](https://github.com/liferay-frontend/liferay-portal/blob/245e40d9ce980034b85ac21f17cb286d4a9a4ea4/modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/app/LiferayApp.js#L207) 

Anyway, `destroyPortlet` should cover more edge : )